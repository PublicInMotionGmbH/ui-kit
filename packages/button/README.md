# Talixo Button

It's just simple wrapper for button.

## How to install

Package is available as `@talixo/button` in NPM registry, so you can use it in your project
using `npm install @talixo/button --save` or `yarn add @talixo/button`.

## Requirements

Your package should additionally have some extra dependencies:

- `@talixo/shared: ^1.0.0-alpha.20`
- `prop-types: ^15.6.1`
- `react: ^16.2.0`
- `react-dom: ^16.2.0`

These packages are required by `@talixo/button`, but you have to install them manually,
to avoid having different versions of these in your application.

## Supported props

It allows any properties which are allowed in `button`. Additionally, it handles some differently:

Property name | Type      | Default | Description                    
--------------|-----------|:-------:|--------------------------------
className     | string    | n/a     | additional class name passed to wrapper
type          | string    | n/a     | button color, one of: `primary`, `secondary`, `tertiary`, `success`, `error`, `info`, `warning`, `link`
ghost         | bool      | `false` | should this button be without solid background?
small         | bool      | `false` | should this button be smaller than default?
fluid         | bool      | `false` | should this button take full width?
submit        | bool      | `false` | should this button have `type` set to `submit`?

## Changelog

- **0.1.0** - initial version
