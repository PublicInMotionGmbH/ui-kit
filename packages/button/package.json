{
  "name": "@talixo/button",
  "version": "1.0.0-alpha.35",
  "description": "UI component for Button",
  "main": "dist/index.js",
  "module": "index.js",
  "jsnext:main": "index.js",
  "scripts": {
    "test": "node ../../scripts/runTests.js --only button",
    "prepublishOnly": "npm run build",
    "build": "node ../../scripts/buildPackages.js --only button"
  },
  "publishConfig": {
    "access": "public"
  },
  "author": "Public in Motion GmbH",
  "repository": {
    "type": "git",
    "url": "https://github.com/PublicInMotionGmbH/ui-kit.git"
  },
  "bugs": {
    "url": "https://github.com/PublicInMotionGmbH/ui-kit/issues"
  },
  "license": "MIT",
  "devDependencies": {
    "@talixo/icon": "^1.0.0-alpha.35",
    "@talixo/shared": "^1.0.0-alpha.35",
    "prop-types": "^15.6.1",
    "react": "^16.2.0"
  },
  "peerDependencies": {
    "@talixo/shared": "^1.0.0-alpha.35",
    "prop-types": "^15.6.1",
    "react": "^16.2.0"
  }
}
