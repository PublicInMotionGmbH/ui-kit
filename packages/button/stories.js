import React from 'react'

import Icon from '@talixo/icon'
import Button from './src/Button'

import { createStoriesFactory, getReadmeDescription } from '@talixo/shared/story'
import { action } from '@storybook/addon-actions'

const readme = getReadmeDescription(require('./README.md'))

const addStory = createStoriesFactory('Button', module)
const click = action('button-click')

addStory('button-default-size', readme, () => (
  <div>
    <h2>Button default</h2>
    <Button onClick={click}>
      Default
    </Button>
    <Button disabled>
      Default Disabled
    </Button>
    <h2>Button primary</h2>
    <Button color='primary' onClick={click}>
      Primary
    </Button>
    <Button color='primary' variant='ghost'>
      Primary Ghost
    </Button>
    <Button color='primary' disabled>
      Primary Disabled
    </Button>
    <h2>Button black</h2>
    <Button color='black'>
      Black
    </Button>
    <Button color='black' variant='ghost'>
      Black Ghost
    </Button>
    <Button color='black' disabled>
      Black Disabled
    </Button>
  </div>
))

addStory('button-small', readme, () => (
  <div>
    <h2>Button default</h2>
    <Button onClick={click} size='small'>
      Default
    </Button>
    <Button size='small' disabled>
      Default Disabled
    </Button>
    <h2>Button primary</h2>
    <Button color='primary' onClick={click} size='small'>
      Primary
    </Button>
    <Button color='primary' variant='ghost' size='small'>
      Primary Ghost
    </Button>
    <Button color='primary' size='small' disabled>
      Primary Disabled
    </Button>
    <h2>Button black</h2>
    <Button color='black' size='small'>
      Black
    </Button>
    <Button color='black' variant='ghost' size='small'>
      Black Ghost
    </Button>
    <Button color='black' size='small' disabled>
      Black Disabled
    </Button>
  </div>
))

addStory('button-full-width', readme, () => (
  <div>
    <Button variant='full-width'>
      Default Full Width
    </Button>
    <Button color='primary' variant='full-width'>
      Primary Full Width
    </Button>
    <Button color='black' variant='full-width'>
      Black Full Width
    </Button>
    <Button color='primary' variant='full-width' disabled>
      Disabled Full Width
    </Button>
  </div>
))

addStory('with icon', readme, () => (
  <div>
    <h2>Default size button with icon</h2>
    <Button>
      <Icon name='home' /> Home
    </Button>
    <Button color='primary'>
      <Icon name='settings' /> Settings
    </Button>
    <Button color='black'>
      <Icon name='cancel' /> Cancel
    </Button>
    <Button disabled>
      <Icon name='location_disabled' /> Disabled
    </Button>

    <h2>Small button with icon</h2>
    <Button size='small'>
      <Icon name='home' /> Home
    </Button>
    <Button color='primary' size='small'>
      <Icon name='settings' /> Settings
    </Button>
    <Button color='black' size='small'>
      <Icon name='cancel' /> Cancel
    </Button>
    <Button size='small' disabled>
      <Icon name='location_disabled' /> Disabled
    </Button>

    <h2>Full width size butoon with icon</h2>
    <Button variant='full-width'>
      <Icon name='home' /> Home
    </Button>
    <Button color='primary' variant='full-width'>
      <Icon name='settings' /> Settings
    </Button>
    <Button color='black' variant='full-width'>
      <Icon name='cancel' /> Cancel
    </Button>
    <Button variant='full-width' disabled>
      <Icon name='location_disabled' /> Disabled
    </Button>
  </div>
))
