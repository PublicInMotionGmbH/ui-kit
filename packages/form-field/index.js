import FormField from './src/FormField'

export {
  FormField
}
