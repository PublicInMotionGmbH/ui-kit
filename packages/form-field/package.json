{
  "name": "@talixo/form-field",
  "version": "1.0.0",
  "description": "UI Component which represents FormField",
  "main": "dist/index.js",
  "module": "index.js",
  "jsnext:main": "index.js",
  "scripts": {
    "test": "node ../../scripts/runTests.js --only form-field",
    "prepublishOnly": "npm run build",
    "build": "node ../../scripts/buildPackages.js --only form-field"
  },
  "author": "Public in Motion GmbH",
  "repository": {
    "type": "git",
    "url": "https://github.com/PublicInMotionGmbH/ui-kit.git"
  },
  "bugs": {
    "url": "https://github.com/PublicInMotionGmbH/ui-kit/issues"
  },
  "license": "MIT",
  "devDependencies" : {
    "@talixo/checkbox": "^1.0.0",
    "@talixo/number-input": "^1.0.0",
    "@talixo/text-input": "^1.0.0"
  },
  "peerDependencies": {
    "prop-types": "^15.6.1",
    "react": "^16.2.0",
    "react-dom": "^16.2.0"
  },
  "dependencies": {
    "@talixo/shared": "^1.0.0"
  }
}
