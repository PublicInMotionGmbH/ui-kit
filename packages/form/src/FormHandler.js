import React from 'react'
import PropTypes from 'prop-types'

import { Formik } from 'formik'

import Form from './Form'
import isField from './isField'
import transformChildrenRecursively from './transformChildrenRecursively'

const propTypes = {
  /** Additional class name for form. */
  className: PropTypes.string,

  /** Error object. Key is a name of a field. */
  errors: PropTypes.objectOf(PropTypes.oneOfType([ PropTypes.arrayOf(PropTypes.string), PropTypes.string ])),

  /** onSubmit callback function. */
  onSubmit: PropTypes.func,

  /** Validation object. A Yup schema or a function that returns a Yup schema. */
  validationSchema: PropTypes.object,

  /** Initial values of form fields. */
  values: PropTypes.object,

  /** Format error messages, passed to Field */
  formatErrorMessage: PropTypes.func,

  /** Component used for Form */
  FormComponent: PropTypes.oneOfType([ PropTypes.func, PropTypes.string ]),

  /** Should disable the form fully? **/
  disabled: PropTypes.bool,

  /** Should the form be read-only fully? **/
  readOnly: PropTypes.bool
}

const defaultProps = {
  FormComponent: Form,
  disabled: false,
  readOnly: false
}

const noop = () => {}

/**
 * Component which represents Form Handler
 *
 * @property {object} props
 * @property {string} [props.className]
 * @property {object} [props.errors]
 * @property {function} [props.formatErrorMessage]
 * @property {function} [props.onSubmit]
 * @property {object} [props.validationSchema]
 * @property {string} [props.values]
 * @property {boolean} [props.disabled]
 * @property {boolean} [props.readOnly]
 *
 * @class
 */
class FormHandler extends React.PureComponent {
  /**
   * Update errors and values if changed from props.
   *
   * @param props
   */
  componentWillReceiveProps (props) {
    if (!this.formik) {
      return
    }

    // Check if any new errors are provided
    if (props.errors !== this.props.errors) {
      this.formik.getFormikBag().setErrors(props.errors || {})
    }

    // Check if any new value is provided
    if (props.values !== this.props.values && props.values != null) {
      const previous = this.props.values || {}
      const bag = this.formik.getFormikBag()

      for (const key in props.values) {
        // User has changed value manually, so do not fill it.
        if (previous[key] !== bag.values[key]) {
          continue
        }

        // Initial value is not changed or next value is empty
        if (props.values[key] === previous[key] || props.values[key] === undefined) {
          continue
        }

        // Update value
        bag.setFieldValue(key, props.values[key])
      }
    }
  }

  /**
   * Transform nodes so they can be recognized by Formik.
   *
   * @param {array} children
   * @param {object} options
   * @param {function} [options.formatErrorMessage]
   * @param {function} options.setFieldValue
   * @param {function} options.handleBlur
   * @param {array} options.values
   * @param {array} options.touched
   * @param {array} options.errors
   *
   * @returns {array}
   */
  buildNodesList (children, options) {
    return transformChildrenRecursively(
      children,
      node => this.transformNode(node, options),
      isField
    )
  }

  /**
   * Transform single node, so they can be recognized by Formik.
   *
   * @param {*} node
   * @param {object} options
   * @param {function} [options.formatErrorMessage]
   * @param {function} options.setFieldValue
   * @param {function} options.handleBlur
   * @param {array} options.values
   * @param {array} options.touched
   * @param {array} options.errors
   *
   * @returns {*}
   */
  transformNode (node, options) {
    const { formatErrorMessage, disabled, readOnly } = this.props
    const { setFieldValue, handleBlur, values, touched, errors } = options
    const { name, onChange, onBlur } = node.props

    // Omit adding node to Formik if it has no name property
    // but add it to children
    if (name == null) {
      return node
    }

    // Modify props of Field
    function _onChange (value) {
      setFieldValue(name, value)

      if (onChange) {
        onChange(value)
      }
    }

    function _onBlur (e) {
      handleBlur({ persist: noop, target: { name } })

      if (onBlur) {
        onBlur(e)
      }
    }

    const nodeProps = {
      ref: node.ref,
      value: values[name],
      error: touched[name] ? errors[name] : null,
      formatErrorMessage: node.props.formatErrorMessage || formatErrorMessage,
      onChange: _onChange,
      onBlur: _onBlur
    }

    if (disabled) {
      nodeProps.disabled = true
    }

    if (readOnly) {
      nodeProps.readOnly = true
    }

    return React.cloneElement(node, nodeProps)
  }

  /**
   * Create form elements from children
   *
   * @param {object} props
   * @returns {React.Element}
   */
  renderForm = props => {
    const { handleSubmit } = props
    const {
      children, onSubmit, onChange, errors,
      initialValues, values, validationSchema,
      FormComponent, disabled, readOnly, ...passedProps
    } = this.props

    if (this.formik && onChange) {
      const nextValues = this.formik.getFormikBag().values

      if (nextValues !== this.values) {
        this.values = nextValues
        onChange(nextValues)
      }
    }

    const elements = this.buildNodesList(children, props)

    return (
      <FormComponent method='POST' onSubmit={handleSubmit} {...passedProps}>
        {elements}
      </FormComponent>
    )
  }

  /**
   * Handle form submitting
   *
   * @param args
   */
  submit = (...args) => {
    if (this.props.onSubmit) {
      this.props.onSubmit(...args)
    }
  }

  /**
   * Save reference to formik
   *
   * @param ref
   */
  saveRef = (ref) => {
    this.formik = ref
    this.values = ref ? ref.getFormikBag().values : null
  }

  render () {
    const {
      children, className, onSubmit, values, disabled, readOnly,
      formatErrorMessage, FormComponent, ...passedProps
    } = this.props

    return (
      <Formik
        {...passedProps}
        initialValues={values}
        ref={this.saveRef}
        render={this.renderForm}
        onSubmit={this.submit}
      />
    )
  }
}

FormHandler.propTypes = propTypes
FormHandler.defaultProps = defaultProps

export default FormHandler
