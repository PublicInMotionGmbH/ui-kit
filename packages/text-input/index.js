import TextInput from './src/TextInput'

export default TextInput
