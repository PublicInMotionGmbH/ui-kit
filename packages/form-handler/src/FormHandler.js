import React from 'react'
import PropTypes from 'prop-types'

import { Formik } from 'formik'

import { FormField } from '@talixo/form-field'

/**
 * Checks if node is a FormField component
 *
 * @param node
 * @returns {boolean}
 */
export function isFormField (node) {
  if (!node || typeof node !== 'object' || typeof node.type !== 'function') {
    return false
  }

  if (node.type === FormField) {
    return true
  }

  const Component = node.type
  // This is required to work properly ith react-hot-loader
  const CallComponent = Component.__reactstandin__getCurrent
    ? Component.__reactstandin__getCurrent()
    : Component

  return CallComponent === FormField
}

const propTypes = {
  errors: PropTypes.object,
  onSubmit: PropTypes.func,
  validationSchema: PropTypes.object,
  initialValues: PropTypes.object
}

const defaultProps = {
  errors: {}
}

/**
 * Component which represents Form Handler
 *
 * @class
 */
class FormHandler extends React.PureComponent {
  /**
   * Update errors and values if changed from props
   *
   * @param props
   */
  componentWillReceiveProps (props) {
    if (!this.formik) {
      return
    }

    // Check if any new errors are provided
    if (props.errors !== this.props.errors) {
      this.formik.getFormikBag().setErrors(props.errors)
    }

    // Check if any new value is provided
    if (props.values !== this.props.values && props.values != null) {
      const previous = this.props.values || {}
      const bag = this.formik.getFormikBag()

      for (const key in props.values) {
        // User has changed value manually, so do not fill it.
        if (previous[key] !== bag.values[key]) {
          continue
        }

        // Initial value is not changed or next value is empty
        if (props.values[key] === previous[key] || props.values[key] === undefined) {
          continue
        }

        // Update value
        bag.setFieldValue(key, props.values[key])
      }
    }
  }

  /**
   * Transform nodes so they can be recognized by Formik
   *
   * @param {*} node
   * @param {object} props
   * @param {function} props.setFieldValue
   * @param {function} props.handleBlur
   * @param {array} props.values
   * @param {array} props.touched
   * @param {array} props.errors
   *
   * @returns {*}
   */
  transformNode = (node, props) => {
    const { setFieldValue, handleBlur, values, touched, errors } = props

    // Return if node is empty or type of string
    if (!node || typeof node !== 'object') {
      return node
    }

    // Check if node is FormField component
    if (isFormField(node)) {
      const name = node.props.name

      // Omit adding node to Formik if it has no name property
      // but add it to children
      if (name == null) {
        return node
      }

      // Modify props of FormField
      return React.cloneElement(node, {
        ref: node.ref,
        value: values[name],
        error: touched[name] ? errors[name] : null,
        onChange: value => setFieldValue(name, value),
        onBlur: () => handleBlur({ persist: () => {}, target: { name: name } })
      })
    }

    // If node is not a FormField component and has no children return node
    if (!node.props.children) {
      return node
    }

    // Check recursively if any child of given node is a FormField component
    if (node.props.children) {
      let isModified = false

      const children = React.Children.map(node.props.children, node => {
        const nextNode = this.transformNode(node, props)

        if (nextNode !== node) {
          isModified = true
        }

        return nextNode
      })

      // if node was modified inside map return this node with modification
      if (isModified) {
        return React.cloneElement(node, {
          ...node.props,
          ref: node.ref
        }, children)
      }
    }

    return node
  }

  /**
   * Create form elements from children
   *
   * @param {object} props
   * @returns {React.Element}
   */
  renderForm = props => {
    const { handleSubmit } = props
    const { children, onSubmit, initialValues, values, validationSchema, ...passedProps } = this.props
    const elements = React.Children.map(children, node => this.transformNode(node, props))

    return (
      <form method='POST' onSubmit={handleSubmit} {...passedProps}>
        {elements}
      </form>
    )
  }

  /**
   * Handle form submitting
   *
   * @param args
   */
  submit = (...args) => {
    if (this.props.onSubmit) {
      this.props.onSubmit(...args)
    }
  }

  /**
   * Save reference to formik
   *
   * @param ref
   */
  saveRef = (ref) => {
    this.formik = ref
  }

  render () {
    const { children, onSubmit, values, ...passedProps } = this.props

    return (
      <Formik
        {...passedProps}
        initialValues={values}
        ref={this.saveRef}
        render={this.renderForm}
        onSubmit={this.submit}
      />
    )
  }
}

FormHandler.propTypes = propTypes
FormHandler.defaultProps = defaultProps

export default FormHandler
