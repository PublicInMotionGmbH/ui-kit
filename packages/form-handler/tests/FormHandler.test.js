import React from 'react'
import { mount } from 'enzyme'

import { Button } from '@talixo/button'
import { FormField } from '@talixo/form-field'
import { TextInput } from '@talixo/text-input'

import FormHandler from '../src/FormHandler'

const createWrapper = (props) => mount(
  <FormHandler {...props}>
    <br />
    <FormField name='test'>
      <TextInput />
    </FormField>
    <div>
      <FormField name='test2'>
        <TextInput />
      </FormField>
    </div>
    <FormField>
      <TextInput />
    </FormField>
    <Button type='submit'>Test</Button>
  </FormHandler>
)

describe('<FormHandler />', () => {
  describe('rendering', () => {
    let wrapper
    beforeEach(() => {
      wrapper = createWrapper()
    })
    afterEach(() => {
      wrapper.unmount()
    })
    it('renders children correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })
    it('should render all children', () => {
      expect(wrapper.find('.talixo-form-field')).toHaveLength(3)
      expect(wrapper.find('br')).toHaveLength(1)
      expect(wrapper.find('.talixo-button')).toHaveLength(1)
    })
  })

  describe('passing props to children', () => {
    let wrapper, spyFn, input
    beforeEach(() => {
      wrapper = createWrapper()
      spyFn = jest.fn()
      input = wrapper.find('.talixo-text-input__input').first()
    })
    afterEach(() => {
      wrapper.unmount()
    })
    it('should invoke setFieldValue when input has changed', () => {
      wrapper.instance().formik.setFieldValue = spyFn
      input.simulate('change', { target: { value: 'My new value' } })
      // expect(spyFn).toHaveBeenCalledTimes(1)
    })

    it('should invoke handleBlur when input has blurred', () => {
      wrapper.instance().formik.handleBlur = spyFn
      input.simulate('focus')
      input.simulate('change', { target: { value: 'My new value' } })
      input.simulate('blur')
      expect(spyFn).toHaveBeenCalledTimes(1)
      // expect(spyFn).toHaveBeenCalledWith({ persist: expect.any(Function), target: { name: 'test' } })
    })
  })

  describe('props updating', () => {
    const initialErrors = { test: null }
    const updatedErrors = { test: 'Test error' }
    const props = { errors: initialErrors }
    const initialValues = { test: 'Test value' }
    const updatedValues = { test: 'Updated test value' }
    const undefValues = { test: undefined }

    describe('errors update', () => {
      let wrapper
      beforeEach(() => {
        wrapper = createWrapper(props)
      })
      afterEach(() => {
        wrapper.unmount()
      })
      it('should update errors inside Formik bag', () => {
        wrapper.setProps({ errors: updatedErrors })
        expect(wrapper.instance().formik.getFormikBag().errors).toEqual(updatedErrors)
      })
    })

    describe('values update', () => {
      let wrapper
      afterEach(() => {
        wrapper.unmount()
      })
      it('should update values inside Formik bag', () => {
        wrapper = createWrapper()
        wrapper.setProps({ values: updatedValues })
        expect(wrapper.instance().formik.getFormikBag().values).toEqual(updatedValues)
      })

      it('should update initial values inside Formik bag', () => {
        wrapper = createWrapper({ values: initialValues })
        wrapper.instance().formik.getFormikBag().setFieldValue('test', 'User changes')
        wrapper.setProps({ values: updatedValues })
        expect(wrapper.instance().formik.getFormikBag().values).toEqual({ test: 'User changes' })
      })

      it('should update keep values inside Formik bagwhen undefined values are passed', () => {
        wrapper = createWrapper({ values: initialValues })
        wrapper.setProps({ values: undefValues })
        expect(wrapper.instance().formik.getFormikBag().values).toEqual(initialValues)
      })
    })
  })

  describe('form submitting', () => {
    let wrapper, form
    const initialValues = { test: 'Test value' }
    const props = { onSubmit: jest.fn(), values: initialValues }
    beforeEach(() => {
      wrapper = createWrapper(props)
      form = wrapper.find('form')
    })
    afterEach(() => {
      wrapper.unmount()
    })
    it('should invoke onSubmit', () => {
      form.simulate('submit')
      expect(props.onSubmit).toHaveBeenCalledWith(initialValues, expect.any(Object))
    })
  })
})
