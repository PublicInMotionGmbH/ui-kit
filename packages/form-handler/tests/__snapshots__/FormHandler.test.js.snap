// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormHandler /> rendering renders children correctly 1`] = `
<form
  errors={Object {}}
  method="POST"
  onSubmit={[Function]}
>
  <br
    key=".0"
  />
  <div
    className="talixo-form-field talixo-form-field--blur"
    name="test"
  >
    <div
      className="talixo-form-field__input-wrapper"
    >
      <div
        className="talixo-text-input talixo-form-field__input talixo-text-input--empty"
      >
        <input
          className="talixo-text-input__input"
          id="form_field_1"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={null}
          type="text"
        />
      </div>
    </div>
  </div>
  <div
    key=".2"
  >
    <div
      className="talixo-form-field talixo-form-field--blur"
      name="test2"
    >
      <div
        className="talixo-form-field__input-wrapper"
      >
        <div
          className="talixo-text-input talixo-form-field__input talixo-text-input--empty"
        >
          <input
            className="talixo-text-input__input"
            id="form_field_2"
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            style={null}
            type="text"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    className="talixo-form-field talixo-form-field--blur"
  >
    <div
      className="talixo-form-field__input-wrapper"
    >
      <div
        className="talixo-text-input talixo-form-field__input talixo-text-input--empty"
      >
        <input
          className="talixo-text-input__input"
          id="form_field_3"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          style={null}
          type="text"
        />
      </div>
    </div>
  </div>
  <button
    className="talixo-button"
    disabled={false}
    type="submit"
  >
    Test
  </button>
</form>
`;
